static/src/javascripts/projects/common/modules/commercial/slice-adverts.js

```diff
@@ -1,120 +1,120 @@
 
-            return Promise.all(chain(adSlices).slice(0, maxAdsToShow).and(map, function ($adSlice, index) {
-                    // When we are inside the AB test we are adding inline1 manually so index needs to start from 2.
-                    var inlineIndexOffset = (config.tests.cmTopBannerPosition) ? 2 : 1;
+                // Slots are inserted every containerGap + 1 slice
+                if (!item.adSlice || index % (containerGap + 1) !== 0) {
+                    return false;
+                }
 
-                    var adName        = 'inline' + (index + inlineIndexOffset),
-                        $mobileAdSlot = bonzo(createAdSlot(adName, 'container-inline'))
-                            .addClass('ad-slot--mobile'),
-                        $tabletAdSlot = bonzo(createAdSlot(adName, 'container-inline'))
-                            .addClass('ad-slot--not-mobile');
+                var containerId  = item.container.getAttribute('data-id');
+                var isContainerClosed = prefs[containerId] === 'closed';
+                var isFrontFirst = isNetworkFront && index === 0;
+                return !(isFrontFirst || isContainerClosed);
+            })
+            // limit to maxAdsToShow
+            .slice(0, maxAdsToShow)
+            // create ad slots for the selected slices
+            .map(function (item, index) {
+                // When we are inside the AB test we are adding inline1 manually so index needs to start from 2.
+                var inlineIndexOffset = (config.tests.cmTopBannerPosition) ? 2 : 1;
+                var adName = 'inline' + (index + inlineIndexOffset);
+                var adSlot = hasFabricAd && index === 0 ?
+                    (fabricAdSlot = createAdSlot('fabric', 'container-inline')) :
+                    createAdSlot(adName, 'container-inline');
 
-                    return new Promise(function (resolve) {
-                        idleFastdom.write(function () {
-                            // add a tablet+ ad to the slice
-                            if (detect.getBreakpoint() !== 'mobile') {
-                                $adSlice
-                                    .removeClass('fc-slice__item--no-mpu')
-                                    .append($tabletAdSlot);
-                            } else {
-                                // add a mobile advert after the container
-                                $mobileAdSlot
-                                    .insertAfter($.ancestor($adSlice[0], 'fc-container'));
-                            }
+                adSlot.className += ' ' + (isMobile ? 'ad-slot--mobile' : 'container-inline');
+
+                return {
+                    anchor: isMobile ? item.container : item.adSlice,
+                    adSlot: adSlot
+                };
+            });
 
-                            resolve(null);
-                        });
+        if (hasFabricAd) {
+            // if there is no fabric creative, we fall back to loading
```

Comment:
(Speaking to Patrick, it seems we're happy with the fabric slot supporting any inline types in addition to Fabric)